packages/web-api-scan-job-manager/src/worker/worker.ts

```diff
@@ -0,0 +1,114 @@
+                if (scanMessages.length === 0) {
+                    this.logger.logInfo(`The storage queue '${this.queue.scanQueue}' has no message to process.`);
+                    if (poolMetricsInfo.load.activeTasks === 0) {
+                        this.logger.logInfo(`Exiting the ${this.jobId} job since there are no active tasks.`);
+                        break;
+                    }
+                }
+
+                if (scanMessages.length > 0) {
+                    tasksQueuedCount = await this.addTasksToJob(scanMessages);
+                }
+            }
+
+            this.poolLoadGenerator.setLastTasksIncrementCount(tasksQueuedCount);
+
+            this.logger.logInfo('Pool load statistics', {
+                activeTasks: poolMetricsInfo.load.activeTasks.toString(),
+                runningTasks: poolMetricsInfo.load.runningTasks.toString(),
+                requestedTasksToAddPerInterval: tasksIncrementCount.toString(),
+                tasksAddedPerInterval: tasksQueuedCount.toString(),
+                samplingIntervalInSeconds: this.poolLoadGenerator.samplingIntervalInSeconds.toString(),
+                processingSpeedTasksPerMinute: this.poolLoadGenerator.processingSpeedPerMinute.toString(),
+                activeToRunningTasksRatio: this.poolLoadGenerator.activeToRunningTasksRatio.toString(),
+            });
+
+            if (this.runOnce) {
+                break;
+            }
+
+            if (moment().toDate() > this.restartAfterTime) {
+                this.logger.logInfo(`Performing scheduled termination after ${this.jobManagerConfig.maxWallClockTimeInHours} hours.`);
+                break;
+            }
+
+            await System.wait(this.jobManagerConfig.addTasksIntervalInSeconds * 1000);
+        }
+    }
+
+    private async getMessages(messagesCount: number): Promise<Message[]> {
+        const messages: Message[] = [];
+        do {
+            const batch = await this.queue.getMessages();
+            if (batch.length === 0) {
+                break;
+            }
+            messages.push(...batch);
+        } while (messages.length < messagesCount);
+
+        return messages;
+    }
+
+    private async addTasksToJob(scanMessages: Message[]): Promise<number> {
+        const jobTasks = await this.batch.createTasks(this.jobId, scanMessages);
+
+        await Promise.all(
+            jobTasks.map(async jobTask => {
+                if (jobTask.state === JobTaskState.queued) {
+                    const message = scanMessages.find(value => value.messageId === jobTask.correlationId);
+                    await this.queue.deleteMessage(message);
+                }
+            }),
+        );
+
+        const jobQueuedTasks = jobTasks.filter(jobTask => jobTask.state === JobTaskState.queued);
+
+        return jobQueuedTasks.length;
+    }
+
+    private async init(): Promise<void> {
+        this.jobManagerConfig = await this.serviceConfig.getConfigValue('jobManagerConfig');
```

Comment:
Is this config a shared config with the other job manager?

Reply:
yep, it should be the same config